### DISCUSSION TITLE
More modern examples?

### DISCUSSION THREAD

**Discussion by VinnyVicious** [DiligentGraphics/DiligentEngine#270] [Ideas]
The Shadows example looks like an example from Horde 3D/Ogre 3D or some other ancient engine. The adaptive tesselation example is too simplistic. Compare it, for example, with the bgfx example: https://bkaradzic.github.io/bgfx/examples.html#tess

In fact, compare with other examples from bgfx. I think Diligent is very capable, but definitely not showing. Or compare to Panda 3D, which has much less capable graphics engine: https://www.panda3d.org/features/ https://github.com/tobspr/RenderPipeline

Also, TheForge samples:
https://github.com/ConfettiFX/The-Forge

Even web engines have juicier examples:
https://threejs.org/examples/#webgl_animation_skinning_ik
https://playground.babylonjs.com/#YX6IB8#229
https://playcanvas.github.io/#/graphics/post-effects

**Comment by TheMostDiligent** [comment]
Note that tutorials (tessellation example in particular that you mentioned) are intended to be simplistic. Their goal is to show one specific feature of the API in a most concise way.
I agree though that there are no high-end rendering examples at the moment. We unfortunately don't have enough capacity to work on examples specifically. At the same time, we are working on some high-level rendering functionality, so we may have something a bit more interesting in the future.

**Comment by VinnyVicious** [comment]
Maybe a showcase of higher-level engines that use Diligent could be used as some sort of showcase or even benchmark?

I know rbfx was ported recently: https://github.com/rbfx/rbfx/pull/590

**Comment by kariem2k** [comment]
What I really like about Diligent is that many of its "samples" are in a form of reusable render passes, and for the rest of the samples are concise and to the point to a large degree. It is far way better bgfx and The forge because their samples are just huge pieces of code and much of these pieces of code are not even related to the example itself. I really like Diligent (with FX) because it sets exactly in the middle between high level APIs and low level APIs, specially it does not force a scene graph on you.
